riemann_client/client.py

```diff
@@ -200,3 +204,143 @@ def send_events(self, events):
     def clear_queue(self):
         """Resets the message/queue to a blank :py:class:`.Msg` object"""
         self.queue = riemann_client.riemann_pb2.Msg()
+
+
+class AutoFlushingQueuedClient(QueuedClient):
+    """A Riemann client using a queue and a timer that will automatically
+    flush its contents if either:
+        - the queue size exceeds :param max_batch_size: or
+        - more than :param max_delay: has elapsed since the last flush and the
+          queue is non-empty.
+
+    if :param stay_connected: is False, then the transport will be
+    disconnected after each flush and reconnected at the beginning of
+    the next flush.
+
+    A message object is used as a queue, and the following methods are given:
+        - :py:meth:`.send_event` - add a new event to the queue
+        - :py:meth:`.send_events` add a tuple of new events to the queue
+        - :py:meth:`.event` - add a new event to the queue from
+           keyword arguments
+        - :py:meth:`.events` - add new events to the queue from dictionaries
+        - :py:meth:`.flush` - manually force flush the queue to the transport
+    """
+
+    def __init__(self, transport, max_delay=0.5, max_batch_size=100,
+                 stay_connected=False):
+        super(AutoFlushingQueuedClient, self).__init__(transport)
+        self.stay_connected = stay_connected
+        self.max_delay = max_delay
+        self.max_batch_size = max_batch_size
+        self.lock = RLock()
+        self.event_counter = 0
+        self.last_flush = time.time()
+        self.timer = None
+
+        # start the timer
+        self.start_timer()
+
+    def connect(self):
+        """Connect the transport if it is not already connected."""
+        if not self.is_connected():
+            self.transport.connect()
+
+    def is_connected(self):
+        """Check whether the transport is connected."""
+        try:
+            # this will throw an exception whenever socket isn't connected
+            self.transport.socket.type
+            return True
+        except (AttributeError, RuntimeError, socket.error):
+            return False
+
+    def event(self, **data):
+        """Enqueues an event, using keyword arguments to create an Event
+
+        >>> client.event(service='riemann-client', state='awesome')
+
+        :param \*\*data: keyword arguments used for :py:func:`create_event`
+        """
+        self.send_events((self.create_event(data),))
+
+    def events(self, *events):
+        """Enqueues multiple events in a single message
+
+        >>> client.events({'service': 'riemann-client', 'state': 'awesome'})
+
+         :param \*events: event dictionaries for :py:func:`create_event`
+         :returns: The response message from Riemann
+        """
+        self.send_events(self.create_event(evd) for evd in events)
+
+    def send_event(self, event):
```

Comment:
This method doesn't need to be duplicated, it's already defined by `QueuedClient`.